auth-service/src/main/kotlin/uk/gov/communities/delta/auth/services/AccessGroupDCLGMembershipUpdateEmailService.kt

```diff
@@ -0,0 +1,160 @@
+package uk.gov.communities.delta.auth.services
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.slf4j.MDCContext
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import uk.gov.communities.delta.auth.config.EmailConfig
+import uk.gov.communities.delta.auth.config.LDAPConfig
+import uk.gov.communities.delta.auth.repositories.LdapUser
+import uk.gov.communities.delta.auth.repositories.searchPaged
+import uk.gov.communities.delta.auth.utils.EmailAddressChecker
+import javax.naming.directory.SearchControls
+import kotlin.coroutines.CoroutineContext
+import kotlin.time.Duration.Companion.seconds
+
+/*
+ * This class handles sending notification emails when users are given membership of the dclg organisation in an access group.
+ * The recipients are the Lead Testers and Dataset Admins for that Access Group, see recipientsForAccessGroup below.
+ * The emails are sent asynchronously from a separate coroutine.
+ * See ticket DT-696 for details.
+ */
+class AccessGroupDCLGMembershipUpdateEmailService(
+    private val ldapServiceUserBind: LdapServiceUserBind,
+    private val ldapConfig: LDAPConfig,
+    private val emailService: EmailService,
+    private val emailConfig: EmailConfig,
+) {
+    private val supervisorJob = SupervisorJob()
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val emailChecker = EmailAddressChecker()
+
+    private val defaultRecipients = emailConfig.dclgAccessGroupUpdateAdditionalRecipients.map {
+        EmailRecipient(
+            it,
+            it.substringBefore('@')
```

Comment:
I feel like I've seen emails get caught in our spam filter for being addressed to e.g. "phoebe.jackson" ? Not entirely sure and can't find any proof but might be worth keeping an eye out for?

Reply:
Ah good to know. This will only be used for the "additional recipients" like DeltaAdmin@levellingup.gov.uk so will hopefully be OK.